### ISSUE TITLE
News - Notes - publication rules improvements

### ISSUE THREAD

**Issue by margondicco** [Meeds-io/MIPs#166]
## Rationale

This note aims to solve the following concerns end-users can experience while publishing articles:
- on the publication drawer, the contrast in the different options is not optimum and the user has difficulty hierarchizing the information
- Administrators or publishers frequently publish articles on specific pages for platform communication. Space publishers may want to highlight this article in their news lists present in their space by republishing it, as it's possible to do it with the activity in the Feed. However, it's not functionally possible now without having permission on the space at the origin of the article.
- Edit the publication of an article will change the position of the article in a news list that already has the article. It generates a disorder in the publication date and the chronology of the News. 

### Top User Stories

**Publish UI improvement**
Some review of the UI publication drawer to make it more readable: 

- Sections titles style review
- Selected target display
  - Display of selected target -> 2 lines = name + description + margin revision + title style review
- Margins between the elements
- "Who will see" option choices UI according to the permission cases
  - Who will see UI review:
      - Restricted audience Only space member: display only the text + tooltip
      - Both audiences: Radio buttons
      - Restricted All user (republishing): display only the text + tooltip

![Image](https://github.com/user-attachments/assets/58d40559-85d5-49dc-b125-d9cb4a1c09bb)

![Image](https://github.com/user-attachments/assets/c5813cf6-f593-47f5-94fb-5d717f4b18f0)

**Republish an article published to all users**
As space publisher/space admin, when someone else publishes an article to "all users" from a space I do not have the publish permission. 
Then I can publish it to my targets without changing the audience or removing the previous targets. 

- I can edit the publication
- I can see the targets I have no permission with the label "Unauthorized target" + tooltip with the date of publication
    - Sorting rule: my selected target/authorized then unauthorized targets
- I cannot remove those I do not have permission for them
- I cannot change the other parameters (post in the feed, hide author, hide reaction)
- I can add other targets
- I can see the audience "All users" but I cannot change it
- I can see the scheduling option if sets but I cannot change it or add it

Nice to have: 
Once the republication is done, the author of the article receives a notification to let them know "User A has republished your article + name of the article in the second line"

![Image](https://github.com/user-attachments/assets/850828b4-f79d-474c-8ef5-2aca3db76e40)

For admin/web contributors, they can do it from any space.

**Edit publishing**
When I edit the publishing to change a parameter of the drawer (post in the feed, add a target, change the audience, remove a target, change hide author or change hide reaction). 

Then
- the publication of the article in the already selected target should not be changed. There is no modification of publication date which means no change of the position in the news list for previously selected targets.
- on hovering over the target I display a tooltip with the publication date 
- the publication of the article in the feed is not updated

**Hide activity**
As space admin/publisher or a member for the article I have written
When I click on a 3 dots of an article activity I have an option "Hide activity". It replaces the "Delete activity". 
Once I click on it the activity is hidden, the option "Post in Feed" in the publication drawer is turn to false, the reaction are not deleted.

If I enable again the "Post in feed" option in the publication drawer, the activity is displayed again at the top of the feed with all the previous reactions. the time stamp is updated to display "Edited less than ..."

**Target administration**
When a target is created manually or automatically at the space creation
Then we should have also as default permission admin + web contributor displayed and it's not possible to remove them.

### Feature Flags

| Property Name  | Default Value | Target Audience | Functional Behaviour |
| ------------- | ------------- | ------------- | ------------- |
|  |   |   |   |

### Access 
   - GUI 
   - API 

### Services & processing
   - Ex: Scheduled Jobs

### Data and persistence
   - Macro data model
   - Persistence layer 
   - Use of caches
## 5. Annexes
   - Reasoning that explain the technical choices
   - Impacts 
   - References

**Comment by Julien-Dubois-eXo** [comment]
@rdenarie could you complete the technical part please?

**Comment by Julien-Dubois-eXo** [comment]
@rdenarie forget my previous message. I forgot to ask the review of @srenault-meeds before. So ready to review the functional part.

**Comment by srenault-meeds** [comment]
Hi,

Thanks. Here are my feedbacks:
> Publish UI Improvements

- Why do we need another MIP for this while you are implementing this drawer in another MIP? (Discussion needed I think)


> As space publisher/space admin, when an article is published to all users from a space I have the publish permission.

- Do you mean an article you have not published?


> I can see the previous target (hidden if I have not permission or displayed if I have permission) + tooltip with the date of publication

- Do you mean "Disabled" by "Hidden"? Indeed, because I can read after this info: "Sorting rule: my selected target then hidden targets"


> I understand that we can publish an article for a new audience. But I cannot edit properties like the author, reactions, etc.

I have two feedbacks about it:
- Do we plan to send a notification to the author to let them know  their article is published somewhere they might not have access to? Goal: inform about it and make sure no confidentiality is broken
- Reactions are related to the space where it is written. So why not reminding it?


> I cannot schedule a publication end date

Why not? I don't understand this one as I can publish into a new target but I can't decide to unpublish it automatically.


> For admin/web contributors, they can do it from any space.

- What can they do actually?


> If I enable again the "Post in feed" option in the publication drawer, the activity is displayed again in the top of the feed with all the previous reactions.

In that case, what date do you consider for the article activity? The initial one with update date or the new one?


> Then we should have also as default permission admin + web contributor displayed and it's not possible to remove them.

- I would say it differently to make sure we understand it well. 
Then admin and web contributors are automatically set as publisher for this target. It is not possible to remove them.

Available for discussion if needed
